toffy/rename_fovs.py

```diff
@@ -0,0 +1,74 @@
+import json
+import os
+from distutils.dir_util import copy_tree
+
+from ark.utils import io_utils
+
+
+def rename_fov_dirs(run_path, fov_dir, new_dir=None):
+    """Renames FOV directories to have specific names sources from the run JSON file
+
+    Args:
+        run_path (str): path to the JSON run file
+        fov_dir (str): directory where the FOV subdirectories are stored
+        new_dir (str): name of new directory to output files to, defaults to None
+
+        """
+
+    io_utils.validate_paths(run_path)
+    io_utils.validate_paths(fov_dir)
+
+    #retieve FOV names and number of scans for each
+    with open(run_path) as f:
+        data = json.load(f)
+        run_data = data['fovs']
+        fov_scan = {x['name']: x['scanCount'] for x in run_data}
```

Comment:
I think it will simplify the logic of the downstream steps if you construct this dictionary to instead be `{current_name: new_name}`. Then you can loop over each sub-folder, and use its current name to get the new name

Reply:
Here's the code adam used to construct the fov-x-scan-y names from the run json:
![image](https://user-images.githubusercontent.com/13770365/163652063-ec0cab42-e645-4b60-acdb-2070f047af37.png)

This will get you the current name of each FOV. For the new names, lets do the following: If there is only one scan of a given fov, then the name is just the user-supplied name (from the `name` field). Otherwise, if there is more than 1 scan, then scans 2-n are named `username-n`.